src/magic.cpp

```diff
@@ -2986,11 +2987,16 @@ spell &known_magic::select_spell( Character &guy )
     } );
 
     uilist spell_menu;
+    float min_y_pixels = EVEN_MINIMUM_TERM_HEIGHT * ImGui::GetTextLineHeight();
     spell_menu.desired_bounds = {
         -1.0,
             -1.0,
-            -1.0,
-            clamp( static_cast<int>( known_spells_sorted.size() ), 24, TERMY * 9 / 10 ) *ImGui::GetTextLineHeight(),
+            std::clamp( float( EVEN_MINIMUM_TERM_WIDTH * ImGui::CalcTextSize( "X" ).x ),
+            ImGui::GetMainViewport()->Size.x * 3 / 8, ImGui::GetMainViewport()->Size.x ),
+            std::max( std::min( float( ( known_spells_sorted.size() + 3 ) * ImGui::GetTextLineHeightWithSpacing() ),
+            ImGui::GetMainViewport()->Size.y * 9 / 10 ),
+            // Limit vertical space at low terminal size; otherwise, give room for descriptions
+            float( ( ImGui::GetMainViewport()->Size.y < min_y_pixels * 1.50 ) ? min_y_pixels : min_y_pixels * 1.50 ) )
```

Comment:
Please move these two calculations to helper methods or something. It's really hard to tell where one stops and the other starts, let alone figuring out it's supposed to be two calculations in the first place.

Reply:
Yeah, it was more legible before the github bot's suggestion... I'll improve it